pkg/trace/api/telemetry.go

```diff
@@ -136,3 +140,14 @@ func (m *telemetryMultiTransport) roundTrip(req *http.Request, endpoint *config.
 	}
 	return resp, err
 }
+
+func ensureContainerID(ctx context.Context, h http.Header) {
+	if id := h.Get(headerContainerID); id != "" {
+		return
+	}
+	if id := GetContainerID(ctx, h); id != "" {
+		h.Set(headerContainerID, id)
+	} else {
+		metrics.Count("datadog.trace_agent.telemetry_proxy.no_container_id_found", 1, []string{}, 1)
+	}
+}
```

Comment:
Unless we're going to use this other places, it may be clearer to just inline it.

Reply:
Sounds good! inlined